packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,365 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { includes, pick } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { useState, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	Notice,
+	PanelBody,
+	RangeControl,
+	ResizableBox,
+	Spinner,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
+import { useViewportMatch } from '@wordpress/compose';
+import {
+	BlockControls,
+	BlockIcon,
+	InspectorControls,
+	MediaPlaceholder,
+	MediaReplaceFlow,
+	__experimentalBlock as Block,
+} from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import icon from './icon';
+import useClientWidth from '../image/use-client-width';
+
+/**
+ * Module constants
+ */
+import { MIN_SIZE } from '../image/constants';
+
+const ALLOWED_MEDIA_TYPES = [ 'image' ];
+const ACCEPT_MEDIA_STRING = 'image/*';
+
+const SiteLogo = ( {
+	alt,
+	attributes: { align, width, height },
+	containerRef,
+	isSelected,
+	setAttributes,
+	url,
+} ) => {
+	const clientWidth = useClientWidth( containerRef, [ align ] );
+	const isLargeViewport = useViewportMatch( 'medium' );
+	const isWideAligned = includes( [ 'wide', 'full' ], align );
+	const isResizable = ! isWideAligned && isLargeViewport;
+	const [ { naturalWidth, naturalHeight }, setNaturalSize ] = useState( {} );
+	const { toggleSelection } = useDispatch( 'core/block-editor' );
+	const classes = classnames( {
+		'is-transient': isBlobURL( url ),
+	} );
+	const { maxWidth, isRTL, title } = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		const siteEntities = select( 'core' ).getEditedEntityRecord(
+			'root',
+			'site'
+		);
+		return {
+			title: siteEntities.title,
+			...pick( getSettings(), [ 'imageSizes', 'isRTL', 'maxWidth' ] ),
+		};
+	} );
+
+	function onResizeStart() {
+		toggleSelection( false );
+	}
+
+	function onResizeStop() {
+		toggleSelection( true );
+	}
+
+	const img = (
+		// Disable reason: Image itself is not meant to be interactive, but
+		// should direct focus to block.
+		/* eslint-disable jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */
+		<a href="#home" className={ classes } rel="home" title={ title }>
+			<span className="custom-logo-link">
+				<img
+					className="custom-logo"
```

Comment:
Do we need a className here? is it for both frontend and editor or just for editor? In both cases, it doesn't follow the guidelines for any of these;

 - If it's frontend and editor it should be `wp-block-site-logo__something` 
 - if it's just for editor it should be `block-library-site-logo__something`

Reply:
The core function `get_custom_logo()` outputs this markup (both the span above and also this class name. I've replicated this in the editor so that the markup between the two is as close as possible.

Reply:
Oh, interesting! it makes sense to keep it as though.